utils/dwarf.c

```diff
@@ -1430,9 +1431,13 @@ static void get_source_location(Dwarf_Die *die, struct build_data *bd,
 		Dwarf_Die cudie;
 		Dwarf_Line *line;
 		unsigned long dwarf_addr = sym_to_dwarf_addr(dinfo, sym->addr);
+		int search_limit = 10;
 
 		dwarf_diecu(die, &cudie, NULL, NULL);
-		line = dwarf_getsrc_die(&cudie, dwarf_addr);
+		do {
+			line = dwarf_getsrc_die(&cudie, dwarf_addr);
+			dwarf_addr += NOP_INSN_SIZE;
+		} while (line == NULL && --search_limit > 0);
```

Comment:
Would be better if we can check the binary and it's NOP actually.  Also it needs to handle the failure case (`line == NULL`) properly so that just skip the function (and `ufftrace replay` would run without the error).

Reply:
Hmm.. I didn't do that because we need load address to read the instruction so it requires map info. Is it possible to get the map info here?

Reply:
Could you please let me know if you have a better idea? Otherwise, I can just add TODO comment in this commit. I think it's better to fix with a workaround than leaving it broken.

Reply:
Agreed, let's fix the problem first.  I think we should check the symbol size as well so that it can stop at the end of the function.

Reply:
Right, I've handled it as you mentioned.